plugins/channel_header_call/channel_header_call.jsx

```diff
@@ -0,0 +1,216 @@
+        if (this.props.dropdownOpen) {
+            activeClass = ' channel-header__icon--active';
+        }
+
+        return (
+            <button
+                id='pluginChannelHeaderButtonDropdown'
+                className={'channel-header__icon channel-header__icon--wide ' + activeClass}
+                type='button'
+                onClick={this.handleClick}
+            >
+                {children}
+            </button>
+        );
+    }
+}
+
+export default class ChannelHeaderCall extends React.PureComponent {
+    static propTypes = {
+
+        /*
+         * Components or actions to add as channel header buttons
+         */
+        components: PropTypes.array,
+
+        channel: PropTypes.object.isRequired,
+        channelMember: PropTypes.object.isRequired,
+
+        /*
+         * Logged in user's theme
+         */
+        theme: PropTypes.object.isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            dropdownOpen: false,
+        };
+    }
+
+    toggleDropdown = (dropdownOpen) => {
+        this.setState({dropdownOpen});
+    }
+
+    onClose = () => {
+        this.toggleDropdown(false);
+    }
+
+    fireActionAndClose = (action) => {
+        action(this.props.channel, this.props.channelMember);
+        this.onClose();
+    }
+
+    createButton = (plug) => {
+        return (
+            <HeaderIconWrapper
+                key={'channelHeaderButton' + plug.id}
+                buttonClass='channel-header__icon style--none'
+                iconComponent={plug.icon}
+                onClick={() => plug.action(this.props.channel, this.props.channelMember)}
+                buttonId={plug.id}
+                tooltipKey={'plugin'}
+                tooltipText={plug.tooltipText ? plug.tooltipText : plug.dropdownText}
+            />
+        );
+    }
+
+    createDropdown = (plugs) => {
+        const items = plugs.map((plug) => {
+            return (
+                <li
+                    key={'channelHeaderPlug' + plug.id}
+                >
+                    <a
+                        href='#'
+                        className='d-flex align-items-center'
+                        onClick={() => this.fireActionAndClose(plug.action)}
+                    >
+                        <span className='d-flex align-items-center overflow--ellipsis'>{plug.icon}</span>
+                        <span>{plug.dropdownText}</span>
+                    </a>
+                </li>
+            );
+        });
+
+        return (
+            <div className='flex-child'>
+                <Dropdown
+                    id='channelHeaderPlugDropdown'
+                    onToggle={this.toggleDropdown}
+                    onSelect={this.onSelect}
+                    open={this.state.dropdownOpen}
+                >
+                    <CustomToggle
+                        dropdownOpen={this.state.dropdownOpen}
+                        bsRole='toggle'
```

Comment:
Is the `bsRole` on these needed? I thought it's only needed if we're using bootstrap.